src/System.Text.Json/tests/Serialization/Value.ReadTests.cs

```diff
@@ -3,6 +3,8 @@
 // See the LICENSE file in the project root for more information.
 
 using System.Collections.Generic;
+using System.Collections.Immutable;
```

Comment:
Can we add some negative tests with unsupported collection types (or artificial types that aren't compatible with our checks). We should be throwing an `InvalidOperationException` I believe (or perhaps NotSupportedException -- need to discuss)

Reply:
I'll follow up with this as part of https://github.com/dotnet/corefx/issues/37520, and following your change adding some more exceptions helpers.